x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/QueryTranslator.java

```diff
@@ -675,6 +680,21 @@ private static Query translateQuery(BinaryComparison bc) {
                 return new RangeQuery(source, name, value, true, null, false, format);
             }
             if (bc instanceof LessThan) {
```

Comment:
Shouldn't a similar query be applied for `LessThanOrEqual`?

Reply:
That's a good question. Now that I am thinking about it and after discussions with @iverase, it looks like it should be applied only to `LessThanOrEqual`, but since the calculation is not precise, maybe we can get away with implementing it for both.